modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/account_contact/edit_address.jsp

```diff
@@ -0,0 +1,208 @@
+							<clay:icon
+								cssClass="reference-mark text-warning"
+								symbol="asterisk"
+							/>
+
+							<span class="hide-accessible sr-only"><liferay-ui:message key="required" /></span>
+						</span>
+					</label>
+
+					<aui:input fieldParam="addressZip" id="addressZip" label="" name="zip" />
+				</div>
+
+				<aui:input cssClass="mailing-ctrl" fieldParam="addressMailing" id="addressMailing" name="mailing" />
+			</clay:sheet-section>
+
+			<clay:sheet-footer
+				cssClass="sheet-footer-btn-block-sm-down"
+			>
+				<div class="btn-group">
+					<div class="btn-group-item">
+						<clay:button
+							displayType="primary"
+							label='<%= LanguageUtil.get(resourceBundle, "save") %>'
+							type="submit"
+						/>
+					</div>
+
+					<div class="btn-group-item">
+						<clay:link
+							cssClass="btn btn-secondary"
+							displayType="null"
+							href="<%= editContactInformationDisplayContext.getRedirect() %>"
+							label='<%= LanguageUtil.get(resourceBundle, "cancel") %>'
+							role="button"
+						/>
+					</div>
+				</div>
+			</clay:sheet-footer>
+		</clay:sheet>
+	</clay:container-fluid>
+
+	<liferay-frontend:component
+		componentId="CountryRegionDynamicSelect"
+		context='<%=
+			HashMapBuilder.<String, Object>put(
+				"countrySelect", portletDisplay.getNamespace() + "addressCountryId"
+			).put(
+				"countrySelectVal", countryId
+			).put(
+				"regionSelect", portletDisplay.getNamespace() + "addressRegionId"
+			).put(
+				"regionSelectVal", regionId
+			).build()
+		%>'
+		module="{CountryRegionDynamicSelect} from account-admin-web"
+	/>
+</aui:form>
+
+<aui:script use="liferay-form">
+	var addressCountry = document.getElementById(
+		'<portlet:namespace />addressCountryId'
+	);
+
+	function checkCountry(countryId) {
+		Liferay.Service(
+			'/country/get-country',
+			{
+				countryId: countryId,
+			},
+			(response, err) => {
+				if (err) {
+					console.error(err);
+				}
+				else {
+					updateAddressZipRequired(response.zipRequired);
+				}
+			}
+		);
+	}
+
+	function handleSelectChange(event) {
+		var value = Number(event.currentTarget.value);
+
+		if (value > 0) {
+			checkCountry(value);
+		}
+		else {
+			updateAddressZipRequired(false);
+		}
+	}
+
+	function updateAddressZipRequired(required) {
+		var addressZipRequiredWrapper = document.getElementById(
+			'<portlet:namespace />addressZipRequiredWrapper'
+		);
+		var formValidator = Liferay.Form.get('<portlet:namespace />fm')
+			.formValidator;
+
+		var rules = formValidator._getAttr('rules');
+
+		if (required) {
+			addressZipRequiredWrapper.removeAttribute('hidden');
+		}
+		else {
+			addressZipRequiredWrapper.setAttribute('hidden', true);
+		}
+
+		rules.<portlet:namespace />addressZip = {required: required};
+	}
+
+	if (addressCountry) {
+		addressCountry.addEventListener('change', handleSelectChange);
+
+		<c:if test="<%= countryId > 0 %>">
+			checkCountry(<%= countryId %>);
+		</c:if>
+	}
+</aui:script>
```

Comment:
Can we put this in a separated js file? I think we are removing the scripts inside jsp